src/stringToPath.ts

```diff
@@ -1,9 +1,21 @@
+type StringToPath<T extends string> = string extends T
+  ? Array<string>
```

Comment:
Is there any value in this type for this case? we can also return `never` here instead so that we "fail up" and users get a compile time error when using this function wrongly.

Reply:
Well I think the question is more "should the type reflect what the function does at runtime _as best as it can_?" vs "should the type protect TS users from accidentally using this function wrongly?"

As it is, if they get the `string[]` inferred, they will already get a compile error if they try to use the result with either `pathOr` or `setPath`, so the "intended" use-cases are protected. 

Returning `never` would imo make sense if we also make the runtime throw/return `undefined` or similar and we disallow "unknown" paths entirely. Right now it at least allows users to just _convert_ dotted paths to tupled paths even if they happen to not care for the strict safety/use it with other safe functions. WDYT?

Reply:
hm. i do think that by far the most common case will be using it with constants, and that's also by far the most useful case. it's true that there's no way in the runtime level to throw if a path is "unknown", but that _is_ what the type system is for

in the case when someone _is_ using a path with variables, then it's not hard to convert it to an array anyway, which can be used with setPath. this is [recommended style when working with lodash](https://github.com/wix-incubator/eslint-plugin-lodash/blob/master/docs/rules/path-style.md) anyway

and it's always easier to expand in the future than contract

Reply:
Alrighty, updated to infer `never` in case a `string` is passed in :)